### PR TITLE
Update response functions to return `MonadAff m => m Response`

### LINKED ISSUE #146: Should functions on `Response` return `MonadAff m => m Response`
I'm curious if functions within `HTTPure.Response` should return `MonadAff m => m Response` rather than `ResponseM` (which is an `Aff Response`).

For example, `ok` would change from:

```purescript
ok :: forall b. Body.Body b => b -> ResponseM
```
to:

```purescript
ok :: forall m b. MonadAff m => Body.Body b => b -> m Response
```

This would allow users working within a monad stack to use `HTTPure.ok` directly rather than `liftAff $ HTTPure.ok`.

Thoughts?

### PR DESCRIPTION
**Pull request by drewolson**
Fixes #146

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ install:
   - wget -O $HOME/purescript.tar.gz https://github.com/purescript/purescript/releases/download/$TAG/linux64.tar.gz
   - tar -xvf $HOME/purescript.tar.gz -C $HOME/
   - chmod a+x $HOME/purescript
-  - npm install -g bower pulp purescript-psa
+  - npm install -g bower pulp purescript-psa psc-package
```
Comment (cprussin) [inline .travis.yml]: Nice, good simple interim solution to fix CI until I stop being lazy and set up a nixjs binary cache :+1:

**Comment by cprussin** [comment]
@drewolson I don't think this will actually allow use of an arbitrary `MonadAff` though because [handleRequest](https://github.com/cprussin/purescript-httpure/blob/master/src/HTTPure/Server.purs#L41-L53) still expects a router that returns a `ResponseM` aka `Aff Response` right?  I think you'll have to actually change everything in `Server.purs` as well to fix that.

Might be worth adding an example/integration test that actually expresses what you want to enable to ensure that this change is sufficient (also would be nice to have the example to reference folks to in general)

**Comment by drewolson** [comment]
@cprussin This does allow arbitrary `MonadAff`s at your _router_ level, as long as your middleware stack handles interpreting those `MonadAff`s back into an `Aff` for `HTTPure`. I updated my commit to include an example of what this enables. Let me try to describe it in words below.

From `HTTPure`'s perspective, it still expects a function that takes an `HTTPure.Request` and returns an `Aff HTTPure.Response`. However, I want to be able to add customer layers to a monadic stack in my middleware. This means that even though the interface to `HTTPure`'s `serve` function remains the same, my actual router (within my middleware stack) is now operating inside of a custom monad stack rather than directly within an `Aff`. With the change in this PR, I can use the `HTTPure.Response` functions directly (e.g. `ok`) without having to `liftAff` everywhere I use them.

To me, this seems like the correct approach. `HTTPure` itself can't (I'd argue shouldn't) handle arbitrary `MonadAff`s at the `serve` level, because then it would need to know how to interpret them. I feel like allowing the user to define these interpreters in the middleware feels right.

Hope that makes sense and that the new example clarifies the use case.

**Comment by cprussin** [comment]
@drewolson would you like a release for this or is it okay to wait until the next time I cut a release?

**Comment by drewolson** [comment]
@cprussin I'm in no rush, take your time on the release.

**Pull request merged**